File: API_MOVIL/API_MOVIL/Models/Document.cs
using System;
using System.Collections.Generic;

namespace API_MOVIL.Models;

public partial class Document
{
    public int Id { get; set; }

    public int? DocumentTypeId { get; set; }

    public string EntityType { get; set; } = null!;

    public int EntityId { get; set; }

    public string FileName { get; set; } = null!;

    public string FilePath { get; set; } = null!;

    public long? FileSizeBytes { get; set; }

    public string? MimeType { get; set; }

    public bool IsEncrypted { get; set; }

    public string? EncryptionKey { get; set; }

    public int UploadedBy { get; set; }

    public DateTime CreatedAt { get; set; }

    public virtual User UploadedByNavigation { get; set; } = null!;
}

File: SIMEXWeb/backend/app/Models/Document.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

/**
 * Documento genérico polimórfico asociado a cualquier entidad del sistema.
 *
 * Se relaciona con la entidad destino mediante el par `(entity_type, entity_id)`
 * (relación morph manual). Soporta cifrado opcional en reposo indicado por
 * `is_encrypted` y la clave en `encryption_key`. Sólo se registra la marca
 * `created_at`; `UPDATED_AT` está desactivado.
 *
 * @property int         $id
 * @property int         $document_type_id FK a `document_types`.
 * @property string      $entity_type      Clase/tipo de la entidad propietaria.
 * @property int         $entity_id        ID de la entidad propietaria.
 * @property string      $file_name        Nombre visible del archivo.
 * @property string      $file_path        Ruta de almacenamiento.
 * @property int         $file_size_bytes  Tamaño en bytes.
 * @property string|null $mime_type
 * @property bool        $is_encrypted     Si el archivo se almacena cifrado.
 * @property string|null $encryption_key   Clave usada para cifrar (si aplica).
 * @property int         $uploaded_by      FK al usuario que lo subió.
 */
class Document extends Model
{
    use HasFactory;

    const UPDATED_AT = null;

    protected $fillable = [
        'document_type_id',
        'entity_type',
        'entity_id',
        'file_name',
        'file_path',
        'file_size_bytes',
        'mime_type',
        'is_encrypted',
        'encryption_key',
        'uploaded_by',
    ];

    /**
     * Casts a tipos PHP nativos.
     *
     * @return array<string, string>
     */
    protected function casts(): array
    {
        return [
            'is_encrypted' => 'boolean',
            'file_size_bytes' => 'integer',
            'created_at' => 'datetime',
        ];
    }

    /**
     * Usuario que subió el documento.
     *
     * @return BelongsTo
     */
    public function uploader(): BelongsTo
    {
        return $this->belongsTo(User::class, 'uploaded_by');
    }
}
